probe-rs/src/session.rs

```diff
@@ -177,12 +193,16 @@ impl Session {
         &self.target.flash_algorithms
     }
 
+    /// Read available data from the SWO interface without waiting.
+    ///
+    /// This method is only supported for ARM-based targets, and will
+    /// return [Error::ArchitectureRequired] otherwise.
     pub fn read_swo(&mut self) -> Result<Vec<u8>, Error> {
         let interface = self.get_arm_interface()?;
         interface.read_swo()
     }
 
-    pub fn get_arm_interface(&mut self) -> Result<&mut Box<dyn ArmProbeInterface>, Error> {
+    fn get_arm_interface(&mut self) -> Result<&mut Box<dyn ArmProbeInterface>, Error> {
```

Comment:
I am not sure if these can be non-public (as in no tool requires them currently). But I guess they should be :)

Reply:
I searched Github, and didn't find any usage of this method. So I assumed that it can be removed.